src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.cs

```diff
@@ -256,5 +256,45 @@ internal static bool Vector128OrdinalIgnoreCaseAscii(Vector128<ushort> vec1, Vec
             // Compare two lowercased vectors
             return (lcVec1 ^ lcVec2) == Vector128<sbyte>.Zero;
         }
+
+        /// <summary>
+        /// Convert Vector128 that represent 8 ASCII UTF-16 characters to lowercase
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static Vector128<ushort> Vector128AsciiToLowercase(Vector128<ushort> vec)
+        {
+            // ASSUMPTION: Caller has validated that input values are ASCII.
+            Debug.Assert(AllCharsInVector128AreAscii(vec));
+
+            // the 0x80 bit of each word of 'lowerIndicator' will be set iff the word has value >= 'A'
+            Vector128<sbyte> lowIndicator1 = Vector128.Create((sbyte)(0x80 - 'A')) + vec.AsSByte();
+
+            // the 0x80 bit of each word of 'combinedIndicator' will be set iff the word has value >= 'A' and <= 'Z'
+            Vector128<sbyte> combIndicator1 = Vector128.LessThan(
+                Vector128.Create(unchecked((sbyte)(('Z' - 'A') - 0x80))), lowIndicator1);
+
+            // Add the lowercase indicator (0x20 bit) to all A-Z letters
```

Comment:
Being picky: `0x20` is a byte not a bit. Maybe
```suggestion
            // Add the lowercase indicator (5th bit, represented by 0x20) to all A-Z letters
```
Same below.

Reply:
I like the original better, fwiw. It's clear to a reader what it means.